roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -90,22 +58,55 @@ def __init__(self, *args, **kwargs):
                 proxies.append(var.upper() + "=" + quote(self.template_var(value)))
         self.skopeo_proxy_vars = " ".join(proxies)
 
-    def is_active(self):
-        """Skip hosts with unsupported deployment types."""
-        deployment_type = self.get_var("openshift_deployment_type")
-        has_valid_deployment_type = deployment_type in DEPLOYMENT_IMAGE_INFO
+    def is_image_local(self, image):
+        """Check if image is already in local docker index."""
+        result = self.execute_module("docker_image_facts", {"name": image})
+        return bool(result.get("images")) and not result.get("failed")
+
+    def local_images(self, images):
+        """Filter a list of images and return those available locally."""
+        found_images = []
+        for image in images:
```

Comment:
`found_images = list(map(self.is_image_local, images))`?

Reply:
This would create a list of booleans?

Reply:
yep, it would run `self.is_image_local` for each item in `images` and create a generator for. Worked fine here

Reply:
This is not what we want.

```sh
>>> a = [1, 2, 0]
>>> def doa(x):
...   return bool(x)
... 
>>> list(map(doa, a))
[True, True, False]
>>> 
```

Reply:
Oh, right, so we want to filter out the list of images which are local. `list(filter(self.is_image_local, images))` should work then.

We could leave it like this though, as its a bit harder to read

Reply:
I prefer to leave as-is.  I didn't define this function, just copy/pasted it and modified some of the other logic bits.